pilot/adapter/config/file/config.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"io/ioutil"
+	"istio.io/istio/pilot/model"
+	"fmt"
+)
+
+const (
+	defaultNamespaceOverride = "default"
+	defaultDomainOverride    = "cluster.local"
+)
+
+// Information for a single element of configuration stored in a file.
+type FileConfig struct {
```

Comment:
here and with all public names in this package starting with `File`, remove the `File` prefix. We refer to this package by its name (`file`) so there's stuttering: `file.FileConfig`. Instead, this should be called `Config` so that at usage sites its `file.Config`, which reads better.